src/Menu/Matcher/Voter/ActiveVoter.php

```diff
@@ -14,23 +14,46 @@
 namespace Sonata\AdminBundle\Menu\Matcher\Voter;
 
 use Knp\Menu\ItemInterface;
+use Knp\Menu\Matcher\Matcher;
 use Knp\Menu\Matcher\Voter\VoterInterface;
 
-/**
- * Active menu voter bases in extra `active`.
- *
- * @final since sonata-project/admin-bundle 3.52
- *
- * @author Samusev Andrey <andrey.simfi@ya.ru>
- */
-class ActiveVoter implements VoterInterface
-{
-    public function matchItem(ItemInterface $item)
+// NEXT_MAJOR: Remove the else part when dropping support of knplabs/knp-menu 2.x
+if (!method_exists(Matcher::class, 'addVoter')) {
```

Comment:
I've used this comparison base on https://github.com/KnpLabs/KnpMenu/compare/2.6.0...v3.0.0#diff-1a1c11c7967a9aa87f7d1ddf4b27eaadL35 to see if it's KnpMenu `2.x` or `3.x`